### ISSUE TITLE
Shortcuts to games are not created outside Flatpak environment

### ISSUE THREAD

**Issue by dariox86** [flathub/net.lutris.Lutris#80]
Desktop and application menu shortcuts are created inside Flatpak environment and they are not visible in the main system. Also I don't think they would work outside Flatpak environment.

**Comment by gasinvein** [comment]
It's currently impossible to create desktop entries from within flatpak apps. See also flatpak/xdg-desktop-portal#326

**Comment by p1u3o** [comment]
> It's currently impossible to create desktop entries from within flatpak apps. See also flatpak/xdg-desktop-portal#326

It should be possible to do --filesystem=~/.local/share/applications etc

I've messed before with using that and up a symlinks etc inside the sandbox and it did successfully generate a desktop file that worked after being modified to use flatpak run.

It's an ugly hack though.

**Comment by gasinvein** [comment]
@p1u3o It isn't that simple. The app has to be aware of flatpak and alter desktop entries accordingly, e.g. `Exec=/some/path` should become `Exec=flatpak run --command=/some/path net.lutris.Lutris`.
Also such an override effectively adds a security hole.

**Comment by p1u3o** [comment]
@gasinvein I mentioned I had to modify the generated .desktop file, but yes, it is a security hole and a terrible idea.

**Comment by FilBot3** [comment]
Somehow the [come.google.Chrome](https://github.com/flathub/com.google.Chrome) Flatpak is able to do this. I'll have to investigate more after work.

**Comment by A6GibKm** [comment]
For future reference xdg-desktop-portal added a portal for this jus tnow, but it is yet to be implemented by any DE.

**Comment by mwleeds** [comment]
> For future reference xdg-desktop-portal added a portal for this jus tnow, but it is yet to be implemented by any DE.

Specifically in xdg-desktop-portal 1.14.0 there's a dynamic launcher portal for creating `.desktop` files outside the sandbox. There are implementations in the xdg-desktop-portal-gtk and xdg-desktop-portal-gnome backends. The gtk one has been released already; the gnome one is on the development branch, not yet released.

https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.DynamicLauncher

**Comment by mwleeds** [comment]
That new portal is also supported by libportal 0.6, in case you don't want to make the D-Bus calls manually.

**Comment by A6GibKm** [comment]
But is the portal supported in, say GNOME 42 or KDE?

**Comment by strycore** [comment]
If lutris has full access to the home folder, can't this be solved without modifying the client by aliasing `lutris` to `flatpak run net.lutris.Lutris`?

**Comment by FilBot3** [comment]
> If lutris has full access to the home folder, can't this be solved without modifying the client by aliasing `lutris` to `flatpak run net.lutris.Lutris`?

I believe you are correct, however, in the interest of security, this would not be advisable. Especially when installing some of these Windows programs on your system. You want them to be sandboxed as much as possible.

**Comment by strycore** [comment]
> I believe you are correct, however, in the interest of security, this would not be advisable. 

I don't understand why, an alias wouldn't change the behavior and sandboxing of a Flatpak. 
Anyway, this is no longer needed since shortcuts created from the flatpak will now call `flatpak run net.lutris.Lutris` instead of `lutris`. If the patch works, then this issue can be closed. (PR #252 needs to be merged before)

**Comment by kon14** [comment]
> I don't understand why, an alias wouldn't change the behavior and sandboxing of a Flatpak. 

It's the home access part they're referring to as not advisable, not the exec alias bit.

Lutris shouldn't be requesting home fs access.

**Comment by strycore** [comment]
> Lutris shouldn't be requesting home fs access.

maybe we can consider removing home access when that won't break any of our features.

**Comment by pm4rcin** [comment]
> > Lutris shouldn't be requesting home fs access.
> 
> maybe we can consider removing home access when that won't break any of our features.

Could you tell which features are broken because from my guessing only `Games` folder is needed in home if at all (because it could be added to persistent)? Also we could just add persistent `.` so it could write to it's own `home`.

**Comment by A6GibKm** [comment]
> > > Lutris shouldn't be requesting home fs access.
> > 
> > 
> > maybe we can consider removing home access when that won't break any of our features.
> 
> Could you tell which features are broken because from my guessing only `Games` folder is needed in home if at all (because it could be added to persistent)? Also we could just add persistent `.` so it could write to it's own `home`.

Lutris is expected to run a library consisting of thousands of games, many of which have the bad habit of requesting permissions from outside their install path.

**Comment by pm4rcin** [comment]
> > > > Lutris shouldn't be requesting home fs access.
> > > 
> > > 
> > > maybe we can consider removing home access when that won't break any of our features.
> > 
> > 
> > Could you tell which features are broken because from my guessing only `Games` folder is needed in home if at all (because it could be added to persistent)? Also we could just add persistent `.` so it could write to it's own `home`.
> 
> Lutris is expected to run a library consisting of thousands of games, many of which have the bad habit of requesting fs access from outside their install path.

So why not change home to `persistent` then? It would close the huge sandbox hole while all various games could still pollute the sandbox directory.

**Comment by strycore** [comment]
We will never remove the home folder from the Lutris flatpak, there's just too much stuff depending on it.

**Comment by pm4rcin** [comment]
> We will never remove the home folder from the Lutris flatpak, there's just too much stuff depending on it.

What about changing it to persistent?

**Comment by strycore** [comment]
that would break integration with Steam and shortcut so this isn't an option.

**Comment by pm4rcin** [comment]
> that would break integration with Steam and shortcut so this isn't an option.

Understood, I'll override it for myself since I don't need the integration or shortcuts. One more question: Does the integration work with Steam flatpak or only native?

**Comment by strycore** [comment]
I think the integration with Steam flatpak is kinda busted.

**Comment by 777Raim** [comment]
Hi there. Has the situation with .desktop files changed? I created the issue on main `lutris` [repo](https://github.com/lutris/lutris/issues/4726), but nobody helped me.

p.s. have `flatpak lutris` version: `0.5.12` and `gnome 43.4` with `wayland`

**Comment by Kalitsune** [comment]
you can run this command to update your icons:
```sh
cp -ans ~/.var/app/net.lutris.Lutris/data/icons/ ~/.local/share/
cp -ans ~/.var/app/net.lutris.Lutris/data/applications/ ~/.local/share/
```
however it's just an hack, and it has a few limitations:
⚠ you'll need to run this command every times you add a new shortcut
⚠ deleting the shortcut using lutris will not delete the symlinks
